src/python/dxpy/sugar/objects.py

```diff
@@ -0,0 +1,439 @@
+        if "recurse" not in kwargs:
+            kwargs["recurse"] = recurse
+        data_obj = dxpy.find_one_data_object(
+            zero_ok=True,
+            more_ok=False,
+            name=name,
+            classname=classname,
+            **kwargs
+        )
+
+    if not data_obj and exists is True:
+        raise dxpy.DXSearchError("File {} does not to exist", data_obj_desc)
+    if data_obj and exists is False:
+        raise dxpy.DXSearchError(
+            "File {} exists but was expected not to exist", data_obj_desc
+        )
+
+    return data_obj
+
+
+def get_workflow(workflow_desc, project=None, **kwargs):
+    """
+    Search for a project workflow or global workflow with the given ID or name.
+
+    Args:
+        workflow_desc: A workflow ID (either 'workflow-xxx' or
+            'globalworkflow-xxx'), name, or DXWorkflow or DXGlobalWorkflow object.
+        project: Project ID or DXProject object. The project to search for the
+            workflow; if not found here it is expected to be a global workflow.
+        kwargs: Additional keyword arguments to use when searching for a workflow
+            or globalworkflow by name.
+
+    Returns:
+        A DXWorkflow or DXGlobalWorkflow object.
+
+    Raises:
+        * dxpy.DXSearchError if the workflow does not exist.
+        * dxpy.exceptions.PermissionDenied if user does not have proper permissions
+            to access the workflow.
+    """
+    if workflow_desc is None:
+        raise ValueError("'workflow_desc' cannot be None.")
+
+    if isinstance(workflow_desc, (dxpy.DXWorkflow, dxpy.DXGlobalWorkflow)):
+        return workflow_desc
+
+    project = get_project(project)
+
+    if resolver.is_hashid(workflow_desc):
+        try:
+            return dxpy.get_handler(workflow_desc, project.get_id())
+        except dxpy.DXError:
+            pass
+
+    workflow_folder, workflow_name, recurse = _parse_object_name(workflow_desc)
+    if "folder" in kwargs:
+        workflow_folder = kwargs.pop("folder")
+    if "recurse" in kwargs:
+        recurse = kwargs.pop("recurse")
+    kwargs["name"] = workflow_name
+    kwargs["return_handler"] = True
+    workflow = dxpy.find_one_data_object(
+        zero_ok=True,
+        more_ok=False,
+        classname="workflow",
+        project=project.get_id(),
+        folder=workflow_folder,
+        recurse=recurse,
+        **kwargs
+    )
+
+    if not workflow:
+        kwargs["limit"] = 2
+        candidates = dxpy.find_global_workflows(**kwargs)
+        result = next(candidates, None)
+        if result is None:
+            raise dxpy.DXSearchError(
+                "Expected one result, but found none: {}".format(str(kwargs))
+            )
+        if next(candidates, None) is not None:
+            raise dxpy.DXSearchError(
+                "Expected one result, but found more: {}".format(str(kwargs))
+            )
+
+    return workflow
+
+
+def get_app_or_applet(app_or_applet_desc, project=None, **kwargs):
```

Comment:
Similar comment to `get_workflow`. This function won't be able to search for Apps.